include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -512,6 +512,26 @@ def XtenNN_SignOp: XTenNN_Op<"sign", [Pure, TosaExtension, ElementwiseUnary, Sam
   let assemblyFormat = [{ operands attr-dict `:` functional-type(operands, results) }];
 }
 
+def XtenNN_TopK: XTenNN_Op<"topk", [
+                            InferTensorTypeAdaptor,
+                            Pure, TosaExtension]> {
+  let summary = "Calculate the topk";
+  let description = [{
+    Follows the specification of ONNX TopK at opset 11
+  }];
+  let arguments = (ins
+    AnyRankedTensor:$input,
+    I64Attr:$k,
```

Comment:
To be consistent with ONNX, this would need to be a Value instead of an Attr. ONNX wants this to be a `tensor<1xi64>`, but an `i64` value would also be fine for me

Reply:
Currently working on that :)